### PR TITLE
chore(sdk): add super experimental csharp client

### PR DESCRIPTION
**Pull request by raubitsj**
Description
-----------
Add experimental dotnet client backed by wandb-core.

How to try:
```
cd experimental/client-csharp/
dotnet build
dotnet run --project examples/Simple
```

TODO:
- [x] Convert to library
- [ ] Clean up output
- [ ] Add header/footer
- [ ] Implement basic mailbox
- [ ] Add a test
- [x] Add an example
- [ ] Package in nuget?
- [ ] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

### CODE REVIEW

**Review by vanpelt — COMMENTED** [review]
🔥

File: experimental/client-csharp/examples/Simple/Program.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿using Wandb.Sdk;
+using System.Collections.Generic;
+
+// See https://aka.ms/new-console-template for more information
+Console.WriteLine("Hello, World!");
+
+var wandb = new WandbClient();
```
Comment (timoffex) [inline experimental/client-csharp/examples/Simple/Program.cs:7]: Could make it `IDisposable` instead and support the `using` syntax, similar to Python's `with`.

File: experimental/client-csharp/examples/Simple/Program.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+catch (Exception ex)
+{
+    // Handle any exceptions that may occur
+    Console.WriteLine($"An error occurred: {ex.Message}");
+}
+finally
+{
+    // Ensure the Finish method is called even if an exception occurs
+    if (wandb != null)
```
Comment (timoffex) [inline experimental/client-csharp/examples/Simple/Program.cs:36]: Can it really be null?

File: experimental/client-csharp/src/Wandb.Sdk/WandbClient.cs [hunk 3]
```diff
@@ -0,0 +1,387 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Collections.Generic;
+using System.Buffers.Binary;
+using Newtonsoft.Json;
```
Comment (timoffex) [inline experimental/client-csharp/src/Wandb.Sdk/WandbClient.cs:10]: I believe JSON conversion is built into .NET now: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/migrate-from-newtonsoft?pivots=dotnet-9-0

**Pull request closed without merging**